src/formatDate.js

```diff
@@ -50,7 +50,39 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const splitedDate = date.split(fromFormat[3]);
+  const dateObject = {};
+  const dateResult = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateObject[fromFormat[i]] = splitedDate[i];
+
+    if (fromFormat[i] === 'YY') {
+      convertingDate(dateObject, splitedDate[i]);
+    }
+  }
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    if (toFormat[i] !== 'YY') {
+      dateResult.push(dateObject[toFormat[i]]);
+    } else {
+      if (toFormat[i].length === 2) {
+        dateResult.push(dateObject.YYYY.slice(2, 4));
+      } else {
+        dateResult.push(dateObject.YYYY);
+      }
```

Comment:
Do we need this `else` statement here?
If you remove it, your code still should work